src/java.desktop/macosx/classes/com/apple/laf/AquaLookAndFeel.java

```diff
@@ -889,7 +889,7 @@ public Object createValue(UIDefaults defaultsTable) {
             "Table.gridColor", white, // grid line color
             "Table.focusCellBackground", textHighlightText,
             "Table.focusCellForeground", textHighlight,
-            "Table.focusCellHighlightBorder", focusCellHighlightBorder,
+            "Table.focusCellHighlightBorder", new BorderUIResource.LineBorderUIResource(deriveContrastFocusRing(selectionBackground), 2),
```

Comment:
Do we also need to do the same for List.focusCellHighlightBorder?

Reply:
@prsadhuk Thank you for reviewing. 

Earlier when I used the default thickness (thickness=1) the cell focus ring was visible but not prominent as shown in the following screenshot. Please let me know if I need to change thickness to 1 or keep it the same?

**Thickness=1**

![image](https://user-images.githubusercontent.com/95945681/158270171-8260c71f-511f-481e-b88e-f226818a8219.png)

The thickness looks the same on retina as well as non-retina display. I have included the screenshots for comparison-

![image](https://user-images.githubusercontent.com/95945681/158270568-b3018b09-8227-4bf0-9751-80d30df4e411.png)

Reply:
Since List.focusCellHighlightBorder uses the same focusRingColor as used by Table we can change it for List as well. Can I make changes to List using this issue and PR or should I track it by adding a new JBS bug/PR ?

Reply:
<img width="1273" alt="Screenshot 2022-03-16 at 1 04 56 PM" src="https://user-images.githubusercontent.com/43534309/158540377-e97cc3c7-64e9-487d-948d-bc1941d5af28.png">

Reply:
Probably to get an identical focus ring, we need to use
[NSSetFocusRingStyle](https://developer.apple.com/documentation/appkit/1473702-nssetfocusringstyle?language=objc)

Reply:
@prsadhuk Thank you for suggesting possible alternatives.

Reply:
There are two different approaches that we can take at this point.

1. Do we go forward with the present changes? (hue and saturation offset, brightness set to 100%) 
2. Keep a common cell focus ring color - Blue as observed in few native apps (Numbers App). In this scenario we keep the cell focus ring color set to blue when selection background color is NOT blue and return a different color when selection background is Blue.

Trying to find the best solution that takes into account native L&F and present Aqua L&F implementations.

Reply:
Following was another suggestion and if we could use the code to produce brighter focus ring color
https://github.com/openjdk/jdk/commit/2f16ca0f59892fead80b209f7819bdbdb713329a

Since the code uses a fixed value for minBrightnessDifference (which determines the returned color),  it did return brighter shade of the background, but it was not prominent/visible against the selection background color for all colors.
The same is true with using Color's brighter() and darker() methods to obtain the Cell Focus Ring color.

The idea behind the present solution is to obtain a cell focus ring color that is not just a brighter shade of selection background but to offset the hue and saturation such that it is prominent but at the same time a color close to the selection background color.

Reply:
@prsadhuk Thank you for mentioning [JDK-8261243](https://bugs.openjdk.java.net/browse/JDK-8261243) issue. It is similar to the JTable cell focus ring. 

If we decide to go with the blue focus ring it would cause the same issue as described in [JDK-8261243](https://bugs.openjdk.java.net/browse/JDK-8261243)

The selectionBackground in JTable matches the accent color. The selectionBackground is a system color that maps to [alternateSelectedControlColor](https://developer.apple.com/documentation/appkit/nscolor/1533135-alternateselectedcontrolcolor) type property. On the right we can see that the selectionBackground color of JTable (which matches the accent color). 

![image](https://user-images.githubusercontent.com/95945681/160025649-565b8e9e-6c32-4823-9971-c7573eec1451.png)

Since the selectionBackground color for JTable is same as the accent color the focus ring won't be visible unless it is **LIGHTER** than accent color as suggested in the [JDK-8261243](https://bugs.openjdk.java.net/browse/JDK-8261243). We could change the present code not to do any hue offsets which would look as follows -

![image](https://user-images.githubusercontent.com/95945681/160028845-6f78f81c-7028-4951-a3c5-aea5beb23fdb.png)

Reply:
I guess "selectionBackground" matches the "highlight" color which seems to follow accent color, by default but we can make highlight color different from accent color, but in this case focus color should match accent color, not highlight color...please confirm if this is what is happening..
Also, alternateSelectedControlColor is deprecated so you should consider replacing with [selectedContentBackgroundColor](https://developer.apple.com/documentation/appkit/nscolor/2998830-selectedcontentbackgroundcolor)

Reply:
@prsadhuk Cell Focus ring color does not change with highlight color settings. Tested for following cases -

<img width="1060" alt="image" src="https://user-images.githubusercontent.com/95945681/160217226-7d4236c7-1d3e-4062-8fad-974a13374624.png">

Reply:
In my testing in osx10.15 with your fix, I see focus ring is following highlight color and not accent color...did you modify anything in your fix?
<img width="688" alt="Screenshot 2022-03-28 at 11 52 09 AM" src="https://user-images.githubusercontent.com/43534309/160338556-117c8e9b-4e78-4368-8040-dfca1d82e1ab.png">

Reply:
@prsadhuk I have pushed latest changes related to AquaLookAndFeel.java file only. The CSystemColors.m is not checked-in because of [selectedContentBackgroundColor](https://developer.apple.com/documentation/appkit/nscolor/2998830-selectedcontentbackgroundcolor) type property build issues. 


> In my testing in osx10.15 with your fix, I see focus ring is following highlight color and not accent color...did you modify anything in your fix? <img alt="Screenshot 2022-03-28 at 11 52 09 AM" width="688" src="https://user-images.githubusercontent.com/43534309/160338556-117c8e9b-4e78-4368-8040-dfca1d82e1ab.png">

For this scenario, wanted to verify if the **selectionBackground** color in AquaLookAndFeel is explicitly set to green?

Reply:
I put a debug log in deriveContrastFocusRing() and it seems to produce this output (for scenario above)

selectedBackgroundColor com.apple.laf.AquaImageFactory$SystemColorProxy[r=54,g=128,b=34]

I see, even if I change the highlight Color to blue, it does not call deriveContrastFocusRing() again and the focus ring remains green although background Color of JTable cell changes to blue

<img width="537" alt="Screenshot 2022-03-29 at 12 49 59 PM" src="https://user-images.githubusercontent.com/43534309/160556234-cf1732f3-57bd-4530-82e6-e2080e9e1551.png">
.

Reply:
I'm currently testing on mac 11.6 and observe that the background color for JTable follows accent color changes (in mac 11.6). Below are screenshots on mac 10.15 vs mac 11.6 under same settings

On Mac 10.15, JTable's background color seems to follow the highlight color changes whereas in 11.6 it follows the accent color changes. In both versions the selectionBackground is obtained from the same property - **alternateSelectedControlColor**

![image](https://user-images.githubusercontent.com/95945681/160733409-612241d0-884d-43d5-9973-4ecd747fb929.png)

The other issue which was related to on-the-fly accent/highlight color changes corresponding to following case - **deriveContrastFocusRing needs to be called every time there is change to System Preference colors** is fixed and the code will be pushed shortly.

Reply:
@prsadhuk Pushed new changes related to -

- On-the-fly focus ring changes
- Since **selectionBackground** was following highlight color changes in mac 10.15 and accent color changes in mac 11.6, I have updated **deriveContrastFocusRing**() to use original **focusRingColor** (which follows accent color changes in both versions) to obtain the lighter version.

Reply:
@prsadhuk I might have missed considering a more generic condition for the edge case. I'll push the line formatting changes, along with the these changes.